### ISSUE TITLE
2-Attend-and-Excite: Attention-Based Semantic Guidance for Text-to-Image Diffusion Models

### ISSUE THREAD

**Issue by mjmaher987** [sharif-ml-lab/visual_compositional_generation_survey#2]
## Task: Read the Paper and Do the following 4 steps
1. Highlight the paper systematically (**green** for important notes - **blue** for very important notes that worth mentioning in the final survey - **red** for the sentences that we don’t understand or need further attention in the future) [In Zotero].
2. Write the Wikipedia summary of the paper in the GitHub issue (_comment in this issue_).
3. Extract the information in the [excel file](https://docs.google.com/spreadsheets/d/15_sh6TqtUvZIJo1-RZxQtsG0Nvg0vPG_IFKCWiGMjdY/edit?usp=sharing).
4. Add the related sentences to the [overleaf file](https://www.overleaf.com/8167246262ccxtrzztjvpz). Note that when you add the sentence, don’t forget to write the reference in the format of ”[ #reference number that is mentioned in the Google Doc - Name of the Author (who added this to the overleaf) ]” at the end of the sentence.

**Comment by AmirMohammadFakhimi** [comment]
# Attend-and-Excite: Attention-Based Semantic Guidance for Text-to-Image Diffusion Models

## Problems
As shall be demonstrated, although Attend-and-Excite explicitly tackles only the issue of **catastrophic neglect**, their solution implicitly encourages **correct bindings between attributes and their subjects**. This can be attributed to the connection between the two issues of catastrophic neglect and attribute binding. The embedding of the text, obtained by a pre-trained text encoder, links information between each subject and its corresponding attributes. For example, in the prompt “a yellow bowl and a blue cat”, the token “cat” receives information from the token “blue” during the text encoding process. Therefore, mitigating catastrophic neglect over the cat should ideally result in enhancing the color attribute (i.e., allowing for correct binding between “cat” and “blue”).

## Solution
They introduce the concept of Generative Semantic Nursing (GSN). In the GSN process, one slightly shifts the latent code at each time step of the denoising process such that the latent is encouraged to better consider the semantic information passed from the input text prompt.

The attention maps define a probability distribution over the text tokens for each image patch, which determines the dominant tokens in the patch. They observe that this text-image interaction is susceptible to neglect. Although each patch can attend freely to all text tokens, there is no mechanism to ensure that all tokens are attended to by some patch in the image. In cases where a subject token is not attended to, the corresponding subject will not be manifested in the output image.

Thus, intuitively, in order for a subject to be present in the generated image, the model should assign at least one image patch to the subject’s token. Attend-and-Excite embodies this intuition by demanding that each subject token is dominant in some patch in the image. They carefully guide the latent at each denoising time step and encourage the model to attend to all subject tokens and strengthen — or excite — their activations. It is on the fly during inference time.

- Attend-and-Excite technique directly optimizes the noised latent, allowing them to synthesize images that vary significantly from those produced by Stable Diffusion.

<img width="527" alt="method" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/79265203/6a7293fe-4662-4c55-bb1c-7743f5f6f70d">

- As you can see in the image, in the second row (Cross Attention box), there is an attention map for each token. You can see the attention map for the *crown* token is not so clear. So the algorithm tries to make it more clear using a new loss function.

## Evaluation
- They construct a new benchmark to evaluate all methods.
- They consider three types of text prompts: (i) “a [animalA] and a [animalB]”, (ii) “a [animal] and a [color][object]”, and (iii) “a [colorA][objectA] and a [colorB ][objectB]”.

- Text-Image Similarities.
    - **Full Prompt Similarity:** For each prompt, we compute the average CLIP cosine similarity between the text prompt and the corresponding set of 64 generated images. We denote this as the **Full Prompt Similarity**.
    - **Minimum Object Similarity:** It is possible the full-text similarity will not capture the existence of neglect. So they split the prompt into two sub-prompts, each containing a single subject (e.g., “a cat”, “a dog”). They then compute the CLIP similarity between each sub-prompt and each generated image. Given the two scores for each image, they are interested in maximizing the smaller of the two as this would correspond to minimizing neglect. They average the smaller of the two scores across all seeds and prompts and denote this as the **Minimum Object Similarity**.

<img width="911" alt="evaluation, image-text similarities" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/79265203/5dd65d98-340b-4067-8153-b01ea240d23f">

- Text-Text Similarities. Given the 64 generated images for a given input prompt, we generate matching image captions using a pre-trained BLIP image-captioning model [Li et al. 2022b]. We then compute the average CLIP similarity between the prompt and all captions. This process is repeated for each subset and the results are averaged across the prompts in the subset.

<img width="454" alt="evaluation, text-text similarities" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/79265203/a1e23662-0139-4099-aa4f-7e9c4de03a5b">

- User Study.

<img width="445" alt="evaluation, user study" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/79265203/16f3430b-ab11-40aa-a661-44a859b785e5">
